File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/wildfire/foundation-to-frame.ts
import { FoundationType } from "./foundation-to-frame/foundation-type";
import { FramingType } from "./foundation-to-frame/framing-type";
import { InspectionOrderWildfireAssessment } from "../wildfire";

export class InspectionOrderWildfireAssessmentFoundationToFrame {
    /*[Parent]*/
    inspectionOrderWildfireAssessment?: InspectionOrderWildfireAssessment;
    Id?: string;
    openingsEmberEntry?: boolean;
    openingsEmberEntryComment?: string;
    elevatedwithCombustibleMaterial?: boolean;
    elevatedwithCombustibleMaterialsComment?: string;
    foundationTypeId?: string;
    // [ForeignKey(nameof(FoundationTypeId))]
    foundationType?: FoundationType;
    foundationComment?: string;
    framingTypeId?: string;
    // [ForeignKey(nameof(FramingTypeId))]
    framingType?: FramingType;

    
    public constructor(init?:Partial<InspectionOrderWildfireAssessmentFoundationToFrame>) {
        Object.assign(this, init);
    }
}

File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderWildfireAssessment/FoundationToFrame.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderWildfireAssessmentFoundationToFrame
    {
        [Parent]
        public InspectionOrderWildfireAssessment InspectionOrderWildfireAssessment { get; set; }
        [Key]
        public Guid Id { get; set; }

        public bool OpeningsEmberEntry { get; set; }
        public string OpeningsEmberEntryComment { get; set; }
        public bool ElevatedwithCombustibleMaterial { get; set; }
        public string ElevatedwithCombustibleMaterialsComment { get; set; }

        public string FoundationTypeId { get; set; }
        [ForeignKey(nameof(FoundationTypeId))]
        public FoundationType FoundationType { get; set; }

        public string FoundationComment { get; set; }

        public string FramingTypeId { get; set; }
        [ForeignKey(nameof(FramingTypeId))]
        public FramingType FramingType { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            // IoWa = Inspection Order Wildfire Assess,ent
            const string entTypeName = "IoWaFoundationToFrame";
            modelBuilder.Entity<InspectionOrderWildfireAssessment>()
                .HasOne(a => a.FoundationToFrame).WithOne(b => b.InspectionOrderWildfireAssessment)
                .HasForeignKey<InspectionOrderWildfireAssessmentFoundationToFrame>()
                .HasConstraintName($"FK_{nameof(InspectionOrderWildfireAssessment)}_{entTypeName}")
                .OnDelete(DeleteBehavior.Cascade);

            var entBuilder = modelBuilder.Entity<InspectionOrderWildfireAssessmentFoundationToFrame>();
            entBuilder.HasOne(a => a.FoundationType).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(FoundationType)}");
            entBuilder.HasOne(a => a.FramingType).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(FramingType)}");
        }
    }
}